xbmc/utils/AMLUtils.cpp

```diff
@@ -484,15 +480,17 @@ bool aml_set_native_resolution(const RESOLUTION_INFO &res, std::string framebuff
 
 bool aml_probe_resolutions(std::vector<RESOLUTION_INFO> &resolutions)
 {
-  std::string valstr, vesastr, dcapfile;
-  dcapfile = CSpecialProtocol::TranslatePath("special://home/userdata/disp_cap");
-
-  if (SysfsUtils::GetString(dcapfile, valstr) < 0)
+  std::string vesastr;
+  std::string dcapfile = CSpecialProtocol::TranslatePath("special://home/userdata/disp_cap");
+  std::string valstr = SysfsUtils::GetString(dcapfile);
+  if (valstr.empty())
   {
-    if (SysfsUtils::GetString("/sys/class/amhdmitx/amhdmitx0/disp_cap", valstr) < 0)
+    std::string valstr = SysfsUtils::GetString("/sys/class/amhdmitx/amhdmitx0/disp_cap");
+    if (valstr.empty())
       return false;
 
-    if (SysfsUtils::GetString("/sys/class/amhdmitx/amhdmitx0/vesa_cap", vesastr) == 0)
+    std::string vesastr = SysfsUtils::GetString("/sys/class/amhdmitx/amhdmitx0/vesa_cap");
+    if (vesastr.length() > 0)
```

Comment:
100% unimportant, but out of curiosity: Why not `!vesastr.empty()`? (considering the check 4 lines above)